ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -1,464 +1,476 @@
       route-target export {{ address_family }} {{ route_target }}
-{%                      endfor %}
-{%                endfor %}
-{%             endif %}
-{%             if router_bgp.vrfs[vrf].router_id is defined %}
+{%                 endfor %}
+{%             endfor %}
+{%         endif %}
+{%         if router_bgp.vrfs[vrf].router_id is arista.avd.defined %}
       router-id {{ router_bgp.vrfs[vrf].router_id }}
-{%             endif %}
-{%             if router_bgp.vrfs[vrf].timers is defined %}
+{%         endif %}
+{%         if router_bgp.vrfs[vrf].timers is arista.avd.defined %}
       timers bgp {{ router_bgp.vrfs[vrf].timers }}
-{%             endif %}
-{%              if router_bgp.vrfs[vrf].networks is defined and router_bgp.vrfs[vrf].networks is not none %}
-{%                  for network in router_bgp.vrfs[vrf].networks | arista.avd.natural_sort %}
-{%                      if router_bgp.vrfs[vrf].networks[network].route_map is defined and router_bgp.vrfs[vrf].networks[network].route_map is not none %}
+{%         endif %}
+{%         for network in router_bgp.vrfs[vrf].networks | arista.avd.natural_sort %}
+{%             if router_bgp.vrfs[vrf].networks[network].route_map is arista.avd.defined %}
       network {{ network }} route-map {{ router_bgp.vrfs[vrf].networks[network].route_map }}
-{%                      else %}
+{%             else %}
       network {{ network }}
-{%                      endif %}
-{%                  endfor %}
-{%              endif %}
-{%             if router_bgp.vrfs[vrf].neighbors is defined and router_bgp.vrfs[vrf].neighbors is not none %}
-{%                 for neighbor in router_bgp.vrfs[vrf].neighbors %}
-{%                      if router_bgp.vrfs[vrf].neighbors[neighbor].remote_as is defined and router_bgp.vrfs[vrf].neighbors[neighbor].remote_as is not none %}
+{%             endif %}
+{%         endfor %}
+{%         if router_bgp.vrfs[vrf].neighbors is arista.avd.defined %}
+{%             for neighbor in router_bgp.vrfs[vrf].neighbors %}
+{%                 if router_bgp.vrfs[vrf].neighbors[neighbor].remote_as is arista.avd.defined %}
       neighbor {{ neighbor }} remote-as {{ router_bgp.vrfs[vrf].neighbors[neighbor].remote_as }}
-{%                      elif router_bgp.vrfs[vrf].neighbors[neighbor].peer_group is defined and router_bgp.vrfs[vrf].neighbors[neighbor].peer_group is not none %}
+{%                 elif router_bgp.vrfs[vrf].neighbors[neighbor].peer_group is arista.avd.defined %}
       neighbor {{ neighbor }} peer group {{ router_bgp.vrfs[vrf].neighbors[neighbor].peer_group }}
-{%                      endif %}
-{%                      if router_bgp.vrfs[vrf].neighbors[neighbor].local_as is arista.avd.defined and router_bgp.vrfs[vrf].neighbors[neighbor].local_as is number %}
+{%                 endif %}
+{%                 if router_bgp.vrfs[vrf].neighbors[neighbor].local_as is arista.avd.defined and router_bgp.vrfs[vrf].neighbors[neighbor].local_as is number %}
```

Comment:
this is the only place where we test for number. I think we should skip that and just run arista.avd.defined.

Reply:
agreed removed